shoop/core/models/addresses.py

```diff
@@ -115,7 +75,34 @@ class Address(NameMixin, models.Model):
     region_code = models.CharField(verbose_name=_('region code'), max_length=16, blank=True)
     region = models.CharField(verbose_name=_('region'), max_length=64, blank=True)
     country = CountryField(verbose_name=_('country'))
-    is_immutable = models.BooleanField(default=False, db_index=True, editable=False, verbose_name=_('immutable'))
+
+    class Meta:
+        abstract = True
+        verbose_name = _('address')
+        verbose_name_plural = _('addresses')
+
+    @classmethod
+    def from_data(cls, data, ignores=()):
+        """
+        Create Address from a data dictionary.
+
+        :type data: dict[str,str]
+        :type ignores: Iterable[str]
+        :param ignores:
+          Fields names to ignore when searching for existing immutable
+          address.
+        :rtype: Address
+        """
+        immutable = (
+            issubclass(cls, ImmutableAddress) or
+            data.pop('immutable', False))
+        for field_name in ignores:
+            data.pop(field_name, None)
```

Comment:
If we fall through to line 105, the newly created address will lack some fields.